maintenance/utils.py

```diff
@@ -122,24 +108,130 @@ def get_config_value(dbinfo, key):
     return ret
 
 
-def fetch_web_log(dbinfo, remote_path, local_path, env):
+def query_connection_info(dbinfo, env):
     """
     Copy the web log file from a remote host to the local host
     for processing
 
     :param dbinfo: DB configuration
-    :param remote_path: path on the remote to copy
-    :param local_path: local path to place the copy
     :param env: dev/tst/ops
+    :return: dict of username, password, host, port
     """
     username = get_config_value(dbinfo, 'landsatds.username')
     password = get_config_value(dbinfo, 'landsatds.password')
-    host = get_config_value(dbinfo, 'url.{}.webtier'.format(env))
+    log_locations = get_config_value(dbinfo, 'url.{}.weblogs'.format(env)).split(',')
+    return {'username': username, 'password': password, 'log_locs': log_locations}
 
-    client = paramiko.SSHClient()
-    client.set_missing_host_key_policy(paramiko.AutoAddPolicy())
 
-    client.connect(host, username=username, password=password, timeout=60)
-    sftp = client.open_sftp()
+def find_remote_files_sudo(user, password, remote_dirs, prefix, port=22):
+    """
+    List files in folder on a remote host which start with a given prefix (`sudo ls`)
+
+    :param user: username to connect as (must have sudo rights)
+    :param password: the password of the user
+    :param port: the port number on the host machine
+    :param remote_dirs: the absolute location of the folder to search (including host)
+    :param prefix: the beginning of all files names to find
+    :return: list of remote full paths
+    """
+    files = []
+    for rloc in remote_dirs:
+        ssh = paramiko.SSHClient()
```

Comment:
Might be able to save keystrokes with this https://github.com/tomerfiliba/plumbum.